variant/tokenizers/delins_base.py

```diff
@@ -0,0 +1,150 @@
+"""A module for DelIns Tokenization Base Class."""
+import re
+from abc import abstractmethod
+from typing import Optional, Dict
+from .tokenizer import Tokenizer
+from variant.schemas.token_response_schema import DelIns, TokenMatchType
+from variant.tokenizers.caches import NucleotideCache
+
+
+class DelInsBase(Tokenizer):
+    """Class for tokenizing DelIns."""
+
+    def __init__(self) -> None:
+        """Initialize the DelIns Base Class."""
+        self.splitter = re.compile(r'delins')
+        self.parts = None
+        self.nucleotide_cache = NucleotideCache()
+
+    def match(self, input_string: str) -> Optional[DelIns]:
+        """Return tokens that match the input string."""
+        if input_string is None:
+            return None
+
+        input_string = str(input_string).lower()
+
+        if input_string.startswith('(') and input_string.endswith(')'):
+            input_string = input_string[1:-1]
+
+        if ('delins' not in input_string) or (
+                (not input_string.startswith('c.') and (not
+                 input_string.startswith('g.')))):
+            return None
+
+        self.parts = {
+            'start_pos_del': None,
+            'end_pos_del': None,
+            'inserted_sequence1': None,
+            'inserted_sequence2': None,
+            'reference_sequence': None
+        }
+
+        parts = self.splitter.split(input_string)
+        self._get_parts(parts)
+
+        # TODO: implement delins range
+        #  Ex: 812_829delins908_925
```

Comment:
create separate issue?

Reply:
I have it in the project meeting notes to ask whether or not Alex wants it implemented now or later. If he wanted it now, I was just going to add in this PR. If he does want it now but reviewing it in a separate PR is easier for you, I can make a new issue!